### PR TITLE
feat: support sending multi collectibles

### PR DESCRIPTION
**Pull request by BalogunofAfrica**
Fixes 
https://github.com/status-im/status-mobile/issues/19143

### Summary
Adds support for ERC1155 (Multi collectibles) send transaction

Adds constants for contract type as seen in status-go https://github.com/status-im/status-go/blob/develop/services/wallet/common/const.go#L31-L38

#### Platforms

- Android
- iOS

##### Functional

- wallet / transactions

### Steps to test

- Open Status
- Go to wallet tab
- Click on account
- Click collectible tab and select an ERC1155 type collectible to send
- Click send and paste recipient wallet address
- Choose the number you want the send by typing and then click continue to get suggested routes
- Once satisfied, slide the "Slide to send" and authorize transaction

https://github.com/status-im/status-mobile/assets/45393944/8c35a8f0-b69b-4454-8618-2eeec951165c

status: ready

### CODE REVIEW

File: src/status_im/contexts/wallet/send/events.cljs
Comment (briansztamfater) [inline src/status_im/contexts/wallet/send/events.cljs]: I know there will be cases where we sent multiple ERC721 tokens in the same multi-transaction, for example batch send for community tokens which are ERC-721 but treated as the same token. Is `:collectible-multi` some defined naming in the wallet domain? If not, I would suggest to just specify the token standard in the tx-type i.e. `:collectible-erc721` or `:collectible-erc1155`, IMO more understandable and extensible naming. WDYT?

Comment (BalogunofAfrica) [reply src/status_im/contexts/wallet/send/events.cljs]: Yeah that makes sense.

**Comment by J-Son89** [comment]
a question, the schema error that pops up in the video - did you address that? 🤔

File: src/status_im/constants.cljs
Comment (J-Son89) [inline src/status_im/constants.cljs]: should we specify what type of contract? 🤔

File: src/status_im/contexts/wallet/send/events.cljs
Comment (J-Son89) [inline src/status_im/contexts/wallet/send/events.cljs]: perhaps we put this set in a constant somewhere?

File: src/status_im/constants.cljs
Comment (BalogunofAfrica) [reply src/status_im/constants.cljs]: That value is for unknown contract: https://github.com/status-im/status-go/blob/develop/services/wallet/common/const.go#L34

Comment (J-Son89) [reply src/status_im/constants.cljs]: I more meant, this is in a very high level file. For that reason contract could mean a plethora of things.
Perhaps smart_contract_... is more appropriate

**Comment by VolodLytvynenko** [comment]
Hi @BalogunofAfrica thank you for PR.  No issues from my side. PR is ready to be merged

**Comment by J-Son89** [comment]
> Hi @BalogunofAfrica thank you for PR. No issues from my side. PR is ready to be merged

@BalogunofAfrica - we have a due process for pr reviews. 
At the very minimum you should have at least one developer approval (ideally giving 24hrs wait time so developers at all timezones have an opportunity to review), if you can wait for 2 approvals it's even better.

After this it makes sense to ask for QA, before is costly of resources as it increases friction on dev changes as it can likely lead to a waste of further QA resources.

I see that this pr is small but it is important we all stick to the process we agreed to so we can keep the quality high.

File: src/status_im/constants.cljs
Comment (BalogunofAfrica) [reply src/status_im/constants.cljs]: I think a prefix of `wallet` also works, just to stay similar to the go side which uses add its under the wallet namespace.

Comment (J-Son89) [reply src/status_im/constants.cljs]: Yep probably better tbh

**Comment by BalogunofAfrica** [comment]
> a question, the schema error that pops up in the video - did you address that? 🤔

Addressed here https://github.com/status-im/status-mobile/pull/20045/commits/b842e0ff7cb9395243c1f189904c9af53fd86709

File: src/status_im/contexts/wallet/send/flow_config.cljs
Comment (ulisesmac) [inline src/status_im/contexts/wallet/send/flow_config.cljs]: Maybe I'm over thinking this, but do we really want to import a function to check this?

The `=` call before is clear enough and simpler since. Also the function name `is-collectible` isn't clear enough, IMO it suggests if an entity (e.g. a map) is a collectible.

cc: @ilmotta wdyt about this practice?

Comment (ilmotta) [reply src/status_im/contexts/wallet/send/flow_config.cljs]: Good question @ulisesmac. I'll just share some thoughts, but I don't see much of a problem with the current change. Up to you guys really.

- It's common practice to isolate certain business rules in functions, even if they're very simple. The advantage of having a separate function like `is-collectible?` is that if the logic to detect a collectible transaction type changes, we'll only need to change the implementation detail, not hunt all places doing equality comparisons. It's a trade-off.
- Some clojurists would try to build systems using more namespaced keywords. For instance, `:collectible` refers to what? `:tx/collectible` gives a different clarity. This to me is the curse also of the tight integration we have with status-go, because we tend to consume values directly from there and we can't properly encode them for idiomatic Clojure due to performance...
- `is-collectible?` is clear to me, but only if I read using the argument `tx-type`. The `is-` prefix is non-idiomatic. Suggestion to rename: `send-utils/tx-type-collectible?`, this way if, in the future, this function is used in places where the argument name is not that clear, the function name will be self-explanatory.
- The implementation of `is-collectible?` could avoid reconstructing the set on every call.

File: src/status_im/contexts/wallet/send/flow_config.cljs
Comment (OmarBasem) [inline src/status_im/contexts/wallet/send/flow_config.cljs:10]: Why are you checking for collectible tx type that way.. you seem to be checking for
```
(def ^:private collectible-tx-set
  #{:tx/collectible-erc-721
    :tx/collectible-erc-1155})
```
if it contains `:collectible`, what is the benefit of that over `(= tx-type :collectible)` ?

Comment (BalogunofAfrica) [reply src/status_im/contexts/wallet/send/flow_config.cljs:10]: The `tx-type` for collectibles used to be just `:collectible` before because we supported only `erc-721` collectibles. But now it is either `:tx/collectible-erc-721` or `:tx/collectible-erc-1155`. The `:collectible` keyword has been renamed to `:tx/collectible-erc-721`.

So a simple `=` for `:collectible` would not suffice, as we have to check if the `tx-type` is either of the 2 supported collectible types.

cc: @ulisesmac

**Review by briansztamfater — APPROVED** [review]
LGTM! Maybe wait for another approval before merging as this PR touches sensible parts of the wallet code

File: src/status_im/contexts/wallet/send/flow_config.cljs
Comment (ulisesmac) [reply src/status_im/contexts/wallet/send/flow_config.cljs]: Thanks for the feedback @ilmotta 

I agree with the rename suggestion and I wasn't aware of the checks inside the helper function.

File: src/status_im/contexts/wallet/send/flow_config.cljs
Comment (ulisesmac) [reply src/status_im/contexts/wallet/send/flow_config.cljs:10]: Totally makes sense for me now. 
Thanks @BalogunofAfrica for the explanation, btw, I think the new name for the function is better :+1:

File: src/status_im/contexts/wallet/send/utils.cljs
Comment (ulisesmac) [inline src/status_im/contexts/wallet/send/utils.cljs:218]: We don't need `contains?`, the set can be used itself as a function to check this, but just saying in case you didn't know, I'm not saying it must be changed :)

**Review by ulisesmac — APPROVED** [review]
Thanks for your PR @BalogunofAfrica, good work! :+1:

File: src/status_im/contexts/wallet/send/events.cljs
Comment (ilmotta) [inline src/status_im/contexts/wallet/send/events.cljs:466]: I haven't reviewed PRs that evolved this `send.events` namespace, but the fact that it has zero coverage is not good. This namespace is not trivial and I'd normally reject as a reviewer. The re-frame architecture was built to help make this layer testable and it's the top-priority recommended by re-frame to test https://github.com/day8/re-frame/blob/5bd82b3d6625af1fac2b21ba0cd5bab448e44ffe/docs/Testing.md#what-to-test.

I know it's not a tech debt from your PR and at this stage it might be too much work to ~attach~ attack this debt. But, it's never too late to play the boy scout rule `Always leave your code cleaner than you found it`.

Perhaps something you guys could figure out in future changes to this `events` namespace.

**Review by ilmotta — APPROVED** [review]

File: src/status_im/contexts/wallet/send/events.cljs
Comment (ulisesmac) [reply src/status_im/contexts/wallet/send/events.cljs:466]: Thanks @ilmotta  I'll consider it for new PRs :)

cc: @status-im/wallet-mobile-devs

**Pull request merged**
